tests/Feature/API/V1/InvestmentProviderConfigApiV1Test.php

```
@@ -0,0 +1,257 @@
+
+    public function test_update_allows_partial_credentials_update(): void
+    {
+        InvestmentProviderConfig::factory()->create([
+            'user_id' => $this->user->id,
+            'provider_key' => 'alpha_vantage',
+            'credentials' => [
+                'api_key' => 'existing-alpha-key',
+            ],
+        ]);
+
+        $response = $this->actingAs($this->user)
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$response'. (undefined)

(UnusedLocalVariable)